### PR TITLE
1906 the_alternate_clustering_solution

### LINKED ISSUE #1906: Cluster map marker icons
We're expecting lots of asset icons in the game.

We may need to cluster them.

This looks like the most active `react` clustering library:
https://www.npmjs.com/package/react-leaflet-cluster

I don't have much experience in Leaflet clustering, so part of this is for @lilitkarapetyan  to explore what is possible, and if necessary, offer options for how clustering is implemented.

I _expect_ this will be a feature within `PlanningForces`.

### PR DESCRIPTION
**Pull request by lilitkarapetyan**
Fixes #1906 

## 🔨Work carried out:

- [x] Tests pass

## Confirmations

- [ ] I have chosen reviewers for my PR.
- [ ] I have assigned myself to this PR.
- [ ] I have chosen an appropriate label for the PR.
- [ ] I have completed the mandatory sections of this document.
- [ ] I have deleted any unused sections.
- [ ] I confirm that I have checked for required README updates and acted accordingly.

### CODE REVIEW

**Comment by IanMayo** [comment]
Thanks @lilitkarapetyan  - it's a good start :-)

I think we also need to `detach` the asset markers.  If I apply a filter to my "My force" I still see the assets originally shown in the table.  I may see new ones, but the ones previously on the table say on the map.

So, on `planning-force` update: I think we need to remove existing markers, and then add the new ones.

File: packages/components/src/local/organisms/planning-force/index.tsx [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
 import { AssetRow } from '../planning-assets/types/props'
 import styles from './styles.module.scss'
+import 'leaflet.markercluster/dist/leaflet.markercluster'
+import 'leaflet.markercluster/dist/MarkerCluster.css'
+import 'leaflet.markercluster/dist/MarkerCluster.Default.css'
 import PropTypes from './types/props'
 
-export const PlanningForces: React.FC<PropTypes> = ({ assets, selectedAssets, setSelectedAssets, interactive }) => {
+const PlanningForces: React.FC<PropTypes> = ({ assets, selectedAssets, setSelectedAssets, interactive }) => {
+  const mcg = L.markerClusterGroup()
```
Comment (IanMayo) [inline packages/components/src/local/organisms/planning-force/index.tsx]: @lilitkarapetyan - does this create `mcg` on each render?  I'm surprised we aren't storing it in some `state-aware` structure so we know it is only created once.

**Comment by IanMayo** [comment]
Any more changes on this @lilitkarapetyan ?  I'm tempted to merge it, and to start learning lessons about how clustering works.

**Comment by lilitkarapetyan** [comment]
> Any more changes on this @lilitkarapetyan ? I'm tempted to merge it, and to start learning lessons about how clustering works.

Yes I added it.

File: packages/components/src/local/organisms/support-mapping/index.tsx [hunk 2]
```diff
@@ -22,6 +23,14 @@ export const SupportMapping: React.FC<PropTypes> = ({
     }
   }, [mapWidth])
 
+  useEffect(() => {
```
Comment (IanMayo) [inline packages/components/src/local/organisms/support-mapping/index.tsx]: @lilitkarapetyan - I don't see what this is doing.  Well, it is called at few times at app loading, but that point there aren't any marker clusters. It isn't called again.

What is it doing?

**Pull request merged**